objectize.cc

```diff
@@ -2,44 +2,49 @@
 #include <hnswlib.h>
 #include <sstream>
 #include <string>
-
+#include <cstddef>
 // typedef size_t labeltype;
-// #include <iostream>
+#include <iostream>
 
 extern "C" {
 
 // memory
 
 EMSCRIPTEN_KEEPALIVE void *doMalloc(size_t size) {
+    std::cout << "Reached function doMalloc " << std::endl;
   return malloc(size);
 }
 
 EMSCRIPTEN_KEEPALIVE void doFree(void *ptr) {
+    std::cout << "Reached function doFree " << std::endl;
   free(ptr);
 }
 
 //
 
-EMSCRIPTEN_KEEPALIVE hnswlib::L2Space *createSpace(
+EMSCRIPTEN_KEEPALIVE hnswlib::CosineSpace *createSpace(
   int dim
 ) {
-  hnswlib::L2Space *space = new hnswlib::L2Space(dim);
+    std::cout << "Reached function createSpace " << std::endl;
+  hnswlib::CosineSpace *space = new hnswlib::CosineSpace(dim);
```

Comment:
Shouldn't we be using the similarity cosign here?

```suggestion
  hnswlib::CosineSimilaritySpace *space = new hnswlib::CosineSimilaritySpace(dim);
```

Reply:
Yea both files are a version of a cosine similarity space I made, so they're similar, CosineSpace was just another attempt.